### ISSUE TITLE
Cannot `opam install nocrypto`

### ISSUE THREAD

**Issue by MassD** [mirleft/ocaml-nocrypto#87]
My system is 

```
$ cat /etc/*-release
NAME="Red Hat Enterprise Linux Server"
VERSION="7.0 (Maipo)"
ID="rhel"
ID_LIKE="fedora"
VERSION_ID="7.0"
PRETTY_NAME="Red Hat Enterprise Linux Server 7.0 (Maipo)"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:redhat:enterprise_linux:7.0:GA:server"
HOME_URL="https://www.redhat.com/"
BUG_REPORT_URL="https://bugzilla.redhat.com/"

REDHAT_BUGZILLA_PRODUCT="Red Hat Enterprise Linux 7"
REDHAT_BUGZILLA_PRODUCT_VERSION=7.0
REDHAT_SUPPORT_PRODUCT="Red Hat Enterprise Linux"
REDHAT_SUPPORT_PRODUCT_VERSION=7.0
Red Hat Enterprise Linux Server release 7.0 (Maipo)
Red Hat Enterprise Linux Server release 7.0 (Maipo)

```

---

opam version is

```
$ opam --version
1.2.2
```

---

```
$ proxy opam install nocrypto
The following actions will be performed:
  - install conf-gmp 1                                  [required by zarith]
  - install zarith   1.4.1                              [required by nocrypto]
  - install nocrypto 0.5.1
===== 3 to install =====
Do you want to continue ? [Y/n] y

=-=- Gathering sources =-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
[nocrypto] Archive in cache
[zarith] Archive in cache

=-=- Processing actions -=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
[ERROR] The compilation of conf-gmp failed at "sh -c cc -c $CFLAGS -I/usr/local/include test.c".

#=== ERROR while installing conf-gmp.1 ========================================#
# opam-version 1.2.2
# os           linux
# command      sh -c cc -c $CFLAGS -I/usr/local/include test.c
# path         /home/cloud-user/.opam/4.02.3/build/conf-gmp.1
# compiler     4.02.3
# exit-code    1
# env-file     /home/cloud-user/.opam/4.02.3/build/conf-gmp.1/conf-gmp-9319-42bb52.env
# stdout-file  /home/cloud-user/.opam/4.02.3/build/conf-gmp.1/conf-gmp-9319-42bb52.out
# stderr-file  /home/cloud-user/.opam/4.02.3/build/conf-gmp.1/conf-gmp-9319-42bb52.err
### stderr ###
# test.c:1:17: fatal error: gmp.h: No such file or directory
#  #include <gmp.h>
#                  ^
# compilation terminated.



=-=- Error report -=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
The following actions were aborted
  - install nocrypto 0.5.1
  - install zarith   1.4.1
The following actions failed
  - install conf-gmp 1
No changes have been performed

=-=- conf-gmp.1 troobleshooting -=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
=> This package relies on external (system) dependencies that may be missing. `opam depext conf-gmp.1' may help you
   find the correct installation for your system.
```

---

Then I did `opam depext conf-gmp.1`

```
$ opam depext conf-gmp.1
# Detecting depexts using flags: x86_64 linux rhel
# No extra OS packages requirements found.

```

---

Also `gmp` is installed.

```
Package 1:gmp-6.0.0-12.el7_1.x86_64 already installed and latest version
Nothing to do

```

**Comment by cfcs** [comment]
The error message stems from inability to find `gmp.h`; the C headers detailing the interface of the compiled GMP library.

You need the libgmp headers, usually `gmp` or `libgmp` with a `-dev` or `-devel` suffix (on RH7 I believe it's `gmp-devel`).

**Comment by MassD** [comment]
@cfcs wow, just installed `gmp-devel`, and everything good now.

Successfully installed

**Comment by hannesm** [comment]
@MassD great! could you tell me what `opam config var os` returns? `conf-gmp` should depend on `gmp-devel` on your os to avoid the next person having the same issue (over [here](https://github.com/ocaml/opam-repository/blob/master/packages/conf-gmp/conf-gmp.1/opam)).

**Comment by MassD** [comment]
```
$ opam config var os
linux
```

**Comment by MassD** [comment]
maybe

```
["linux"] ["gmp-devel"]

```

**Comment by hannesm** [comment]
sorry, what does `opam depext --flags` return?

**Comment by MassD** [comment]
```
$ opam depext --flags
# Depexts flags detected on this system: x86_64 linux rhel
```

**Comment by hannesm** [comment]
i moved bug to https://github.com/ocaml/opam-depext/issues/32 (we'll need to get depext fixed...)

**Comment by avsm** [comment]
This should be fixed shortly in depext 0.9.1, and then we just need to add the right `rhel` depexts to opam-repository (which should be the same as the `centos` ones).
